doc/query.md

```diff
@@ -76,6 +81,54 @@ if (name == null) {
 
 If an index needs to be changed, first delete the existing index by calling `deleteIndexNamed(String indexName)` where the argument is the index name, then call  `ensureIndexed(List<Object> fieldNames, String indexName)` with the new definition.
 
+#### Managing a Text Index
+
+In order to perform text searches on documents in a datastore a text index needs to be created.  Our text indexing process leverages SQLite's full text search (FTS) functionality, therefore SQLite FTS needs to be available for a text index to be created.
+
+The method `isTextSearchEnabled()` which is part of the index manager API provides a way to check whether text search is currently enabled or disabled.  Text search is deemed as enabled once the presence of the `ENABLE_FTS3` and `ENABLE_FTS3_PARENTHESIS` SQLite compile options is confirmed.  If text search is not enabled see [compiling and enabling SQLite FTS][enableFTS] for details on enabling FTS in SQLite.
+
+[enableFTS]: http://www.sqlite.org/fts3.html#section_2
+
+Managing a text index is not much different from managing a JSON index.  However, to create a text index use `ensureIndexed(List<Object> fieldNames, String indexName, String indexType)` and provide an index type of `text`.
+
+```java
+if (isTextSearchEnabled()) {
+    // Create a text index over the name and comment fields.
+    String name = im.ensureIndexed(Arrays.<Object>asList("name", "comment"), 
+                                   "basic_text_index", 
+                                   "text");
+    if (name == null) {
+        // there was an error creating the index
+    }
+}
+```
+
+By default, the "simple" tokenizer that comes standard with SQLite is used when building the text index if no index settings are provided.  You can override this default tokenizer by specifying a different tokenizer as part of the index settings.  To do this use `ensureIndexed(List<Object> fieldNames, String indexName, String indexType, Map<String, String> indexSettings)` and set indexSettings to a Map containing the `tokenize` key and the name of the desired tokenizer.
+
+```java
+if (isTextSearchEnabled()) {
+    Map<String, String> tokenizer = new HashMap<String, String>();
+    tokenizer.add("tokenize", "porter");
```

Comment:
Call this variable `settings`, otherwise this snippet implies that the last argument is specifically tokenizer.